### PR TITLE
Add removeChatPattern

### LINKED ISSUE #1877: Add a way to delete chat patterns
## Is your feature request related to a problem? Please describe.
Yes, If I have a pattern for invalid input into chat and valid input into chat, once one is fulfilled, I don't care about the other anymore. I'd like to invalidate the other one before it can cause problems later in my program.

## Describe the solution you'd like
`bot.deleteChatPattern(chatPatternName)` & `bot.deleteChatPatternSet(chatPatternName)` (should be aliases of the same thing)

## Describe alternatives you've considered
none that I can think of

## Additional context
N/A

### PR DESCRIPTION
**Pull request by BlueBurgersTDD**
Makes it possible to remove chat patterns
Closes #1877

### CODE REVIEW

**Comment by BlueBurgersTDD** [comment]
Should I make it throw a error when you try to remove a pattern that doesn't exist?

File: test/externalTests/chat.js [hunk 1]
```diff
@@ -60,6 +60,19 @@ module.exports = () => {
...
+    bot.removeChatPattern('test')
+    let triggered = false
+    const listener = () => { triggered = true }
+    bot.once('chat:test', listener)
+    bot.chat('/tellraw @p {"translate":"chat.type.text", "with":["U9G", "Hello"]}')
+    await once(bot, 'message')
+    assert.ok(triggered === false)
+    bot.off('chat:test')
+  })
+
```
Comment (BlueBurgersTDD) [inline test/externalTests/chat.js:75]: This seems like it doesn't pass, hmmm

**Comment by BlueBurgersTDD** [comment]
Ready for review

File: docs/api.md [hunk 2]
```diff
@@ -1418,6 +1418,16 @@ the event will be called `"chat:name"`, with name being the name passed
   * `repeat` - defaults to true, whether to listen for this event after the first match
   * `parse` - instead of returning the actual message that was matched, return the capture groups from the regex
 
+#### bot.removeChatPattern(name)
+
+removes a chat pattern
+* `name` the name of the chat pattern
+
+#### bot.removeChatPatternSet(name)
```
Comment (extremeheat) [inline docs/api.md:1426]: Am I mistaken or do you have two APIs that do the same thing?

Comment (BlueBurgersTDD) [reply docs/api.md:1426]: They do the same thing but I can remove 1

**Comment by BlueBurgersTDD** [comment]
Ready for review

**Review by extremeheat — APPROVED** [review]

**Review by rom1504 — APPROVED** [review]

**Pull request merged**